full-stack-tests/core/src/frontend/hub/ITwinPlatformEnv.ts

```diff
@@ -34,32 +34,42 @@ export class IModelBankFrontend implements TestFrontendHubAccess {
     this._hubClient = new IModelBankClient(orchestratorUrl, undefined);
   }
 
-  public async getLatestChangesetId(arg: IModelIdArg): Promise<ChangesetId> {
+  private async _getChangesetFromId(arg: IModelIdArg & { changeSetId: string }): Promise<ChangesetIndexAndId> {
+    const changeSets: ChangeSet[] = await this._hubClient.changeSets.get(arg.accessToken, arg.iModelId, new ChangeSetQuery().byId(arg.changeSetId));
+    if (!changeSets[0] || !changeSets[0].index || !changeSets[0].changeSetId)
+      throw new BentleyError(BentleyStatus.ERROR, `Changeset ${arg.changeSetId} not found`);
+    return { index: +changeSets[0].index, id: changeSets[0].changeSetId };
+  }
+
+  public async getLatestChangeset(arg: IModelIdArg): Promise<ChangesetIndexAndId> {
     const changeSets: ChangeSet[] = await this._hubClient.changeSets.get(arg.accessToken, arg.iModelId, new ChangeSetQuery().top(1).latest());
-    return (changeSets.length === 0) ? "" : changeSets[changeSets.length - 1].wsgId;
+    if (!changeSets[0] || !changeSets[0].index || !changeSets[0].id)
+      throw new BentleyError(BentleyStatus.ERROR, `No changesets found for ${arg.iModelId}`);
+    return { index: +changeSets[0].index, id: changeSets[0].id };
   }
 
-  public async getChangesetIdFromVersion(arg: IModelIdArg & { version: IModelVersion }): Promise<ChangesetId> {
+  public async getChangesetFromVersion(arg: IModelIdArg & { version: IModelVersion }): Promise<ChangesetIndexAndId> {
     const version = arg.version;
     if (version.isFirst)
-      return "";
+      return { index: -1, id: "" };
```

Comment:
doesn't index 0 mean first "before first changeset" (what we now call "version0")?

Reply:
good catch; fixed.

Reply:
i think it should be 0, not 1.

Reply:
Well, that's embarrassing 😅, sorry. Should actually be fixed now